Foundation/NSString.swift

```diff
@@ -1214,7 +1214,17 @@ extension NSString {
     }
     
     public convenience init(format: String, locale: AnyObject?, arguments argList: CVaListPointer) {
-        NSUnimplemented()    
+        let str: CFString
+        if let loc = locale {
+            if loc.dynamicType === NSLocale.self || loc.dynamicType === NSDictionary.self {
+                str = CFStringCreateWithFormatAndArguments(kCFAllocatorSystemDefault, unsafeBitCast(loc, to: CFDictionary.self), format._cfObject, argList)
+            } else {
+                fatalError("locale parameter must be a NSLocale or a NSDictionary")
+            }
+        } else {
+            str = CFStringCreateWithFormatAndArguments(kCFAllocatorSystemDefault, nil, format._cfObject, argList)
+        }
+        self.init(str._swiftObject)
```

Comment:
Technically on darwin the only reason why you can pass a swift class to this function is because they have root objc inheritance (that has the right methods implemented). However you cannot pass a struct to it even if it does conform to the descriptionWithLocale or the CustomStringConvertible protocol and I doubt there is any real way we can get struct passing to va_args to work in swift on either platform for formatting.

As it stands this implementation looks like a good start, if you want to make a bug on bugs.swift.org and have the format callbacks to be a later followup that is perfectly reasonable.